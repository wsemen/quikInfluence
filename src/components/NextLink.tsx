import { Link as ChakraLink, LinkProps } from '@chakra-ui/layout';
import Link from 'next/link';
import React from 'react';

export type NextLinkProps = Omit<LinkProps, 'as'> & {
  href: string;
  as?: string;
  isInternal?: boolean;
  onClick?: () => void;
  hardRefresh?: boolean;
};

const NextLink: React.FC<NextLinkProps> = ({
  href,
  isInternal = true,
  onClick = () => {},
  as,
  children,
  hardRefresh = false,
  ...rest
}) => {
  if (isInternal) {
    return hardRefresh ? (
      <ChakraLink href={href} onClick={onClick} {...rest}>
        {children}
      </ChakraLink>
    ) : (
      <Link href={href} as={as} passHref>
        <ChakraLink onClick={onClick} {...rest}>
          {children}
        </ChakraLink>
      </Link>
    );
  }

  let finalHref: string;
  const pathToUrl = as ?? href;
  if (/^(https|http|mailto)/.test(pathToUrl)) {
    finalHref = pathToUrl;
  } else {
    finalHref = `/${pathToUrl}`;
  }

  return (
    <ChakraLink href={finalHref} onClick={onClick} {...rest}>
      {children}
    </ChakraLink>
  );
};

export default NextLink;
