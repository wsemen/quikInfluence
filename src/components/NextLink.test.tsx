import { render } from '@testing-library/react';
import NextLink from './NextLink';

it('renders NextLink component', () => {
  const home = render(<NextLink href={'/#'} />);
  expect(home.container).toMatchSnapshot();
  const http = render(
    <NextLink isInternal={false} href={'https://www.google.com'} />
  );
  expect(http.container).toMatchSnapshot();
  const dashboard = render(<NextLink hardRefresh href={'/dashboard'} />);
  expect(dashboard.container).toMatchSnapshot();
  const notInternal = render(
    <NextLink isInternal={false} href={'dashboard'} />
  );
  expect(notInternal.container).toMatchSnapshot();
});
