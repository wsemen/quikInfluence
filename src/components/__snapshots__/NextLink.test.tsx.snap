// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NextLink component 1`] = `
<div>
  <a
    class="chakra-link css-0"
    href="/#"
  />
</div>
`;

exports[`renders NextLink component 2`] = `
<div>
  <a
    class="chakra-link css-0"
    href="https://www.google.com"
  />
</div>
`;

exports[`renders NextLink component 3`] = `
<div>
  <a
    class="chakra-link css-0"
    href="/dashboard"
  />
</div>
`;

exports[`renders NextLink component 4`] = `
<div>
  <a
    class="chakra-link css-0"
    href="/dashboard"
  />
</div>
`;
